src/pages/workspace/WorkspaceInvitePage.js

```diff
@@ -83,19 +83,51 @@ function WorkspaceInvitePage(props) {
     const excludedUsers = useMemo(() => PolicyUtils.getIneligibleInvitees(props.policyMembers, props.personalDetails), [props.policyMembers, props.personalDetails]);
 
     useEffect(() => {
-        const inviteOptions = OptionsListUtils.getMemberInviteOptions(props.personalDetails, props.betas, searchTerm, excludedUsers);
-
-        // Update selectedOptions with the latest personalDetails and policyMembers information
-        const detailsMap = {};
-        _.forEach(inviteOptions.personalDetails, (detail) => (detailsMap[detail.login] = OptionsListUtils.formatMemberForList(detail)));
-        const newSelectedOptions = [];
-        _.forEach(selectedOptions, (option) => {
-            newSelectedOptions.push(_.has(detailsMap, option.login) ? {...detailsMap[option.login], isSelected: true} : option);
+        let emails = searchTerm.replace(/\s,\s/g, ',').split(',');
+        emails = _.filter(
+            _.map(emails, (word) => word.trim()),
+            (email) => email !== '',
+        );
+
+        const newUsersToInviteDict = {};
+        const newPersonalDetailsDict = {};
+        const newSelectedOptionsDict = {};
+
+        _.forEach(emails, (email) => {
+            const inviteOptions = OptionsListUtils.getMemberInviteOptions(props.personalDetails, props.betas, email, excludedUsers);
+
+            // Update selectedOptions with the latest personalDetails and policyMembers information
+            const detailsMap = {};
+            _.forEach(inviteOptions.personalDetails, (detail) => (detailsMap[detail.login] = OptionsListUtils.formatMemberForList(detail)));
+
+            const newSelectedOptions = [];
+            _.forEach(selectedOptions, (option) => {
+                newSelectedOptions.push(_.has(detailsMap, option.login) ? {...detailsMap[option.login], isSelected: true} : option);
+            });
+
+            const userToInvite = inviteOptions.userToInvite;
+
+            // Only add the user to the invites list if it is valid
+            if (userToInvite) {
+                newUsersToInviteDict[userToInvite.accountID] = userToInvite;
+            }
+
+            // Add all personal details to the new dict
+            _.forEach(inviteOptions.personalDetails, (details) => {
+                newPersonalDetailsDict[details.accountID] = details;
+            });
+
+            // Add all selected options to the new dict
+            _.forEach(newSelectedOptions, (option) => {
+                newSelectedOptionsDict[option.accountID] = option;
+            });
         });
 
-        setUserToInvite(inviteOptions.userToInvite);
-        setPersonalDetails(inviteOptions.personalDetails);
-        setSelectedOptions(newSelectedOptions);
+        // Strip out dictionary keys and update arrays
+        setUsersToInvite(_.map(newUsersToInviteDict, (v) => v));
```

Comment:
Any reason why we can't use `_.values(newUsersToInviteDict)` instead here? It would feel more "natural".

Reply:
Yeah that is much nicer, wish I knew about it earlier! Will do that now.